### PR TITLE
BFB porting ice_nucleation to cpp

### PR DESCRIPTION
**Pull request by xyuan**
Porting 
ice_nucleation 
to scream

BFB

### CODE REVIEW

**Comment by xyuan** [comment]
Hopefully, this PR will works also,

sizeof(Real) 8 avx  packsize 1 compiler GCC FPE off #threads 1
Comparing with test8 at tol 0.0e+00
--- checking case # 1, it = 1/1 ---
--- checking case # 2, it = 1/6 ---
--- checking case # 2, it = 2/6 ---
--- checking case # 2, it = 3/6 ---
--- checking case # 2, it = 4/6 ---
--- checking case # 2, it = 5/6 ---
--- checking case # 2, it = 6/6 ---
--- checking case # 3, it = 1/1 ---
--- checking case # 4, it = 1/6 ---
--- checking case # 4, it = 2/6 ---
--- checking case # 4, it = 3/6 ---
--- checking case # 4, it = 4/6 ---
--- checking case # 4, it = 5/6 ---
--- checking case # 4, it = 6/6 ---

File: components/scream/src/physics/p3/p3_functions_ice_nucleation_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ 
+   const auto t_lt_icenuct = temp < icenuct;
+   const auto supi_ge_005 = supi >= 0.05;
+   
+   const auto any_if_log     = t_lt_icenuct && supi_ge_005 && log_predictNc;
+   const auto any_if_not_log = t_lt_icenuct && supi_ge_005 && (!log_predictNc);
+
+   Spack dum{0.0}, N_nuc{0.0}, Q_nuc{0.0};
+
+   dum.set(any_if_not_log,
```
Comment (ambrad) [inline components/scream/src/physics/p3/p3_functions_ice_nucleation_impl.hpp]: It's better not to use `set` for temporaries. Just write `auto dum = ...`. Only the output variables require masked assignments.

**Comment by xyuan** [comment]
Nice, this one passed again,
All Jobs Finished; status = PASSED

File: components/scream/src/physics/p3/p3_functions_ice_nucleation_impl.hpp (hunk 1 above)
Comment (xyuan) [reply components/scream/src/physics/p3/p3_functions_ice_nucleation_impl.hpp]: changes are made based on your comments

**Pull request merged**